### ISSUE TITLE
Reading a part of an HDF5Compound dataset

### ISSUE THREAD

**Issue by tamasgal** [JuliaIO/HDF5.jl#427]
I ended up using a flat data structure in our data format, so instead of having thousands of `/event/1`, `/event/2`, etc. we now have a single table and a second table which holds `event_index` and `event_length`, so we can directly access the data in the flat `/events` dataset.
This is now really fast in Python but I could not figure out a way to read in parts of an HDF5Compound dataset with Julia.

So something like

```
f["/hits"][100:250]
Dataset indexing (hyperslab) is available only for bits types
```

fails obviously. Now I am a bit confused: is this even possible with the current implementation of HDF5.jl?

**Comment by damiendr** [comment]
I'd also like to have this feature. I might give it a go — did this prove problematic or is it just not implemented yet?

**Comment by damiendr** [comment]
In particular, what's up with this:
https://github.com/JuliaIO/HDF5.jl/blob/c04c6206a885474f7ad99878d89649c68401e780/src/HDF5.jl#L2137-L2138
Is there a particular reason why one shouldn't depend on the high-level APIs?

**Comment by crstnbr** [comment]
@damiendr Did you make any progress on this? Are there any half-way attempts? I will give this a try but want to get as much info/help as possible first.

**Comment by damiendr** [comment]
@crstnbr not on the indexing specifically. I did write some code a while ago that sits on top of HDF5.jl and reads compound types much faster: https://github.com/damiendr/HDFTables.jl
It's not yet registered as a package or tested on Julia 1.0 (I'll get to that some time this month).

This uses `HDF5.read_array()` which in turns calls `h5d_read()` with `H5S_ALL` as the dataspace. Hopefully you would only have to modify that call and supply a dataspace object for the slice, but I haven't tried it.

**Comment by crstnbr** [comment]
Turns out that JLD now allows hyperslabbing (is this a word?) in `getindex` of a `JldDataset`:

```julia
jldopen("myfile.jld", "r") do f
    f["my_dataset"][4] # only read 4th element
end
```

Strangely, `read(f["my_dataset"], 4)` doesn't work for me though. Anyways, this should be sufficient for my use case. I'll take a look at your package nonetheless!

**Comment by kleinhenz** [comment]
closed by #652.
